x-pack/legacy/plugins/task_manager/task_store.ts

```diff
@@ -137,6 +138,39 @@ export class TaskStore {
     return savedObjectToConcreteTaskInstance(savedObject);
   }
 
+  /**
+   * Reschedules a previously scheduled task.
+   *
+   * @param task - The task being rescheduled.
+   */
+  public async reschedule(
+    taskInstanceScheduling: TaskInstanceScheduling
+  ): Promise<ConcreteTaskInstance> {
+    const taskInstance = await this.getTask(taskInstanceScheduling.id);
+    return await this.update(
+      applyConcreteTaskInstanceDefaults(
+        taskInstance.status === 'idle'
+          ? {
+              ...omit(taskInstance, 'runAt'),
+              ...taskInstanceScheduling,
```

Comment:
nit: One thing I keep catching myself with in TypeScript is passing objects through. For example, those who call taskManager.reschedule(...) from JavaScript files are able to pass any parameter beyond `id`, `runAt` and `interval` and make it through to the update call. We could deconstruct the parameters in the function definition above `{ id, runAt, interval }`.